File: arrays-js/largestnumbertwiceofothers.js
function largestNumberTwiceOfOthers(nums) {
  let max = Math.max(...nums);                  // O(n) 
  let maxIndex = nums.indexOf(max);             // O(n) 
  for (let i = 0; i < nums.length; i++) {       // O(n) 
    if (i !== maxIndex && nums[i] * 2 > max) {  // O(1)
      return -1;                                // O(1)
    }
  }
    return maxIndex;
}

// Time: O(n)
// Space: O(1)
var res = largestNumberTwiceOfOthers([3, 6, 1, 0])    // 1
console.log(res);

File: arrays/LargestNumberTwiceOfOthers.java
public class LargestNumberTwiceOfOthers {
    public int dominantIndex(int[] nums) {
        int max = 0;
        int maxIndex = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > max) {
                max = nums[i];
                maxIndex = i;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if (i != maxIndex && nums[i] * 2 > max) {
                return -1;
            }
        }
        return maxIndex;
    }

    public static void main(String[] args) {
        LargestNumberTwiceOfOthers obj = new LargestNumberTwiceOfOthers();
        System.out.println(obj.dominantIndex(new int[] { 3, 6, 1, 0 }));
        System.out.println(obj.dominantIndex(new int[] { 1, 2, 3, 4 }));
    }
}
